Modules/cjkcodecs/_codecs_jp.c

```diff
@@ -611,8 +611,19 @@ ENCODER(shift_jis_2004)
                             if (code == DBCINV)
                                 return 1;
                             }
-                            else
+                            else if (ch2 != 0) {
                                 insize = 2;
+                            }
+                            else {
+                                /* Don't consume null char as part of pair */
+                                code = find_pairencmap(
+                                    (ucs2_t)c, 0,
+                                    jisx0213_pair_encmap,
+                                    JISX0213_ENCPAIRS);
+                                if (code == DBCINV) {
+                                    return 1;
+                                }
+                            }
```

Comment:
Don't we already checked `find_pairencmap(c, ch2 /* == 0 */, ...)`?

```suggestion
                            else if (ch2 != 0) {
                                /* Don't consume null char as part of pair */
                                insize = 2;
                            }
```